tictactoe.rb

```diff
@@ -1,184 +1,185 @@
-	end
-	attr_reader :player_turn
+		attr_reader :player_turn
 
-	def turn
-		puts "#{@player_turn}'s turn."
-		puts "Enter your move: "
-		
+		def turn
+			puts "#{@player_turn}'s turn."
+			puts "Enter your move: "
+			
 
-	end
+		end
 
-	def switch_turn
-		if @player_turn == "X"
-			@player_turn = "O"
-		elsif @player_turn == "O"
-			@player_turn = "X"
+		def switch_turn
+			if @player_turn == "X"
+				@player_turn = "O"
+			elsif @player_turn == "O"
+				@player_turn = "X"
+			end
 		end
-	end
 
-	def start_game
-		puts "Welcome to Tic Tac Toe"
-		puts "Play by typing the column position and row position separated by a space"
-		puts "Ex:  top middle"
-		puts "\"#{@player_turn}\" goes first."
+		def start_game
+			puts "Welcome to Tic Tac Toe"
+			puts "Play by typing the column position and row position separated by a space"
+			puts "Ex:  top middle"
+			puts "\"#{@player_turn}\" goes first."
 
+		end
 	end
-end
 
-class Board
+	class Board
 
-	def initialize
-		# @board = [["-","-","-"],["-","-","-"],["-","-","-"]]
-		@board = [["X","O","X"],["O","X","X"],["O","X","-"]]
-	end
+		def initialize
+			# @board = [["-","-","-"],["-","-","-"],["-","-","-"]]
+			@board = [["X","O","X"],["O","X","X"],["O","X","-"]]
+		end
 
-	def display
-		print_line = ""
-		@board.each do |row|
-			row.each do |element|
-				print_line += element + " "
+		def display
+			print_line = ""
+			@board.each do |row|
+				row.each do |element|
+					print_line += element + " "
+				end
+				print_line += "\n"
 			end
-			print_line += "\n"
+			puts print_line
 		end
-		puts print_line
-	end
 
-	def make_move(who_player)
-    board_idx = condition_input
-    while !(valid_move?(board_idx[0], board_idx[-1]))
-			# if valid_move?(board_idx[0], board_idx[1])
-			# 	@board[board_idx[0]][board_idx[1]] = who_player
-			# else
-			board_idx = condition_input
-			valid_move?(board_idx[0], board_idx[-1])
+		def make_move(who_player)
+	    board_idx = condition_input
+	    while !(valid_move?(board_idx[0], board_idx[-1]))
+				# if valid_move?(board_idx[0], board_idx[1])
+				# 	@board[board_idx[0]][board_idx[1]] = who_player
+				# else
+				board_idx = condition_input
+				valid_move?(board_idx[0], board_idx[-1])
+			end
+			@board[board_idx[0]][board_idx[-1]] = who_player
 		end
-		@board[board_idx[0]][board_idx[-1]] = who_player
-	end
 
-	def condition_input
-		move = gets.chomp
-		move_a = move.split(' ')
-		row_word = move_a[0]
-		column_word = move_a[1]
-		return space(row_word, column_word)
-	end
-
-	def space(row, column)
-		case row
-		when "top"
-			first = 0
-		when "middle"
-			first = 1
-		when "bottom"
-			first = 2
-		else
-			first = 3
+		def condition_input
+			move = gets.chomp
+			move_a = move.split(' ')
+			row_word = move_a[0]
+			column_word = move_a[1]
+			return space(row_word, column_word)
 		end
 
-		case column
-		when "left"
-			second = 0
-		when "middle"
-			second = 1
-		when "right"
-			second = 2
-		else
-			second = 3
+		def space(row, column)
```

Comment:
This function is ripe for optimization. This is super verbose and very repetitive. I am not saying this is bad, and I haven't spent the time to think about how to improve this but it really appears like a good candidate. See if you can refactor this to be a bit better.

Reply:
Also formatting please.

Reply:
replace the case statements with a hash that matches string positions with integer values